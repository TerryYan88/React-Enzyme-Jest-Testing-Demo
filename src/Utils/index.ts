/* eslint-disable react/forbid-foreign-prop-types */
import {ShallowWrapper} from "enzyme";
import checkPropTypes from "check-prop-types";
import { applyMiddleware,createStore } from "redux";
import { rootReducer } from "../redux/reducers/rootReducer";


export const testStore = (initialState={}) =>{
    const createStoreWithMiddleware = applyMiddleware()(createStore)
    return createStoreWithMiddleware(rootReducer,initialState);
}


export const findByTestAttr = (component: ShallowWrapper,attr:string)=>{
    const wrapper = component.find(`[data-test="${attr}"]`);
    return wrapper;
}

export const checkProps = (component:any,expectedProps:{})=>{
    const propsErr = checkPropTypes(component.propTypes,expectedProps,'props',component.name);
    return propsErr;
}


