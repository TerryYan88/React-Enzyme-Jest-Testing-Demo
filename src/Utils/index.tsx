
import {ShallowWrapper} from "enzyme";


export const findByTestAttr = (component: ShallowWrapper,attr:string)=>{
    const wrapper = component.find(`[data-test="${attr}"]`);
    return wrapper;
}