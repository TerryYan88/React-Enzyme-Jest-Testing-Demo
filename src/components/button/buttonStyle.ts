import styled from "styled-components";

export const BtnContainer = styled.button`
    border:none;
    width:200px;
    padding:12px 6px;
    background-color:pink;
    font-size:24px;
    font-weight:500;
    border-radius:10px;
`
