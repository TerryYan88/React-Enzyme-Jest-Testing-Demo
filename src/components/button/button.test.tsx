import React from "react";
import renderer from "react-test-renderer";
import {shallow,ShallowWrapper,} from "enzyme";
import SharedButton,{ButtonProps} from "./button";
import {BtnContainer} from "./buttonStyle";
import {findByTestAttr,checkProps} from "../../Utils/index";

const setUp = (props:ButtonProps)=>{
    const component = shallow(<SharedButton {...props}/>) // can test props 
    return component;
}

describe("Button Component",()=>{
    it("should show a button", () => {
        const props:ButtonProps = {
            buttonText:"Example Button Text",
            emitEvent:()=>{}
        };
        const component = (<SharedButton {...props}/>);
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
      })
    describe("Checking PropTypes",()=>{
        it("Should Not Throw a warning",()=>{
            const expectedProps:ButtonProps = {
                buttonText:"Example Button Text",
                emitEvent:()=>{}
            };
            const propsError = checkProps(SharedButton,expectedProps);
            expect(propsError).toBeUndefined();
        })
    })
    describe("Renders",()=>{
        let wrapper:ShallowWrapper
        beforeEach(()=>{
            const props:ButtonProps = {
                buttonText:"Example Button Text",
                emitEvent:()=>{}
            }
            wrapper = setUp(props)
        })

        it("Should render a button",()=>{
            const btnContainer = wrapper.find(BtnContainer);
            expect(btnContainer.length).toBe(1);
            const button = findByTestAttr(wrapper,"buttonComponent");
            expect(button.length).toBe(1);
        })
        
    })

})