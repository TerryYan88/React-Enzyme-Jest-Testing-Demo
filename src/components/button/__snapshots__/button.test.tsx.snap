// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should show a button 1`] = `
.c0 {
  border: none;
  width: 200px;
  padding: 12px 6px;
  background-color: pink;
  font-size: 24px;
  font-weight: 500;
  border-radius: 10px;
}

<button
  className="c0"
  data-test="buttonComponent"
  onClick={[Function]}
>
  Example Button Text
</button>
`;
